bgpd/bgp_evpn.c

```diff
@@ -1670,8 +1670,9 @@ static inline bool bgp_evpn_route_add_l3_ecomm_ok(struct bgpevpn *vpn,
 {
 	return p->prefix.route_type == BGP_EVPN_MAC_IP_ROUTE
 	       && (is_evpn_prefix_ipaddr_v4(p)
-		   || !IN6_IS_ADDR_LINKLOCAL(
-			   &p->prefix.macip_addr.ip.ipaddr_v6))
+		   || (is_evpn_prefix_ipaddr_v6(p) ? !IN6_IS_ADDR_LINKLOCAL(
+			       &p->prefix.macip_addr.ip.ipaddr_v6)
+						   : false))
```

Comment:
please change- 
(is_evpn_prefix_ipaddr_v6(p) ? !IN6_IS_ADDR_LINKLOCAL(&p->prefix.macip_addr.ip.ipaddr_v6) : false)
to -
(is_evpn_prefix_ipaddr_v6(p) && !IN6_IS_ADDR_LINKLOCAL(&p->prefix.macip_addr.ip.ipaddr_v6))

does the same and IMO more readable

Reply:
That was what I had drafted up first, but thought I could avoid an unnecessary call of IN6_IS_ADDR_LINKLOCAL when there's no ip6 address by doing it this way.  Happy to make the change though :-)